test/vm/kernels.jsonl

```diff
@@ -0,0 +1,12 @@
+{"headers": ["http://vault.centos.org/centos/6.9/os/x86_64/Packages/kernel-devel-2.6.32-696.el6.x86_64.rpm"], "kernels": ["http://vault.centos.org/centos/6.9/os/x86_64/Packages/kernel-2.6.32-696.el6.x86_64.rpm"]}
+{"headers": ["http://mirror.centos.org/centos/7.9.2009/updates/x86_64/Packages/kernel-devel-3.10.0-1160.49.1.el7.x86_64.rpm"], "kernels": ["http://mirror.centos.org/centos/7.9.2009/updates/x86_64/Packages/kernel-3.10.0-1160.49.1.el7.x86_64.rpm"]}
+{"headers": ["http://amazonlinux.us-east-1.amazonaws.com/2/core/2.0/x86_64/3cea6107e9fffdc60f8ca99c7dac421b427d2ea06003cf5eb970344e4c2e18f4/../../../../../blobstore/13b711e5117ca3ad3c459a303dedfdb6dc0ad7c1440aabf02df6407a43bd94df/kernel-devel-4.14.299-223.520.amzn2.x86_64.rpm"], "kernels": [""]}
```

Comment:
@FedeDP changing URL to get the kernels RPMs work for everything in the kernel crawler output except for AmazonLinux2 and the other newer AmazonLinux ones and we would need those to make the test robust. Would you have ideas re how to retrieve the correct kernel RPM, the URL is quite strange ...

Reply:
> Would you have ideas re how to retrieve the correct kernel RPM, the URL is quite strange ...

Well, I expected `kernel`, or `linux`, or `linux-image` to be present; but they are not...

A `yum search` run on amazonlinux 2023 ec2 instance gives:
```
sudo yum search kernel
Last metadata expiration check: 0:52:55 ago on Mon Jun  5 12:32:10 2023.
========================================================================================= Name & Summary Matched: kernel =========================================================================================
kernel.x86_64 : The Linux kernel
kernel-devel.x86_64 : Development package for building kernel modules to match the kernel
kernel-headers.x86_64 : Header files for the Linux kernel for use by glibc
```

but headers and kernel are not there...

Reply:
Well, it seems like some `kernel`s are there; i updated `kernel-crawler` amazonlinux2 impl to return all deps, and it returned:
```
...
{
"kernelversion": 1,
"kernelrelease": "4.19.84-33.70.amzn2.x86_64",
"target": "amazonlinux2",
"headers": [
"http://amazonlinux.us-east-1.amazonaws.com/2/extras/kernel-ng/stable/x86_64/f9bb600b110c667c6e2c57454f861bb81462dc01d366747d6614565287e7f066/../../../../../../blobstore/87c3cff795dd362a9788483354a0e82165136ef6cf851df525f749414ca7f3de/kernel-devel-4.19.84-33.70.amzn2.x86_64.rpm"
]
},
{
"kernelversion": 1,
"kernelrelease": "4.19.84-33.70.amzn2.x86_64",
"target": "amazonlinux2",
"headers": [
"http://amazonlinux.us-east-1.amazonaws.com/2/extras/kernel-ng/stable/x86_64/f9bb600b110c667c6e2c57454f861bb81462dc01d366747d6614565287e7f066/../../../../../../blobstore/dcfb3c9186b315da366f3d82ac77bdc9d77b86020cc1c10a5262546225249511/kernel-4.19.84-33.70.amzn2.x86_64.rpm"
]
},
{
"kernelversion": 1,
"kernelrelease": "4.19.72-25.58.amzn2.x86_64",
"target": "amazonlinux2",
"headers": [
"http://amazonlinux.us-east-1.amazonaws.com/2/extras/kernel-ng/stable/x86_64/f9bb600b110c667c6e2c57454f861bb81462dc01d366747d6614565287e7f066/../../../../../../blobstore/652cee0fc7c4dd8036aa5e6488a20277bdb7a4ffcaa896b9748ddb9110dd5979/kernel-devel-4.19.72-25.58.amzn2.x86_64.rpm"
]
},
{
"kernelversion": 1,
"kernelrelease": "4.19.72-25.58.amzn2.x86_64",
"target": "amazonlinux2",
"headers": [
"http://amazonlinux.us-east-1.amazonaws.com/2/extras/kernel-ng/stable/x86_64/f9bb600b110c667c6e2c57454f861bb81462dc01d366747d6614565287e7f066/../../../../../../blobstore/907a972944ed2eb1b7e58793c399f398520729ac6b17506d037acb5c2eeec371/kernel-4.19.72-25.58.amzn2.x86_64.rpm"
]
}
]
```
and so on.

Reply:
Also,
```
grep "4.14.299" out.txt
"kernelrelease": "4.14.299-223.520.amzn2.x86_64",
"http://amazonlinux.us-east-1.amazonaws.com/2/core/2.0/x86_64/e5756b881dd10f379f337f77c404dbc4c557dcbacab9b1b8a3aea0fb859ea0fb/../../../../../blobstore/01ff4f0b1a4f8a68cd1fdf8723581bf3eeb5e70cb31cc49294e33ec841d5e09f/kernel-4.14.299-223.520.amzn2.x86_64.rpm"
"kernelrelease": "4.14.299-223.520.amzn2.x86_64",
"http://amazonlinux.us-east-1.amazonaws.com/2/core/2.0/x86_64/e5756b881dd10f379f337f77c404dbc4c557dcbacab9b1b8a3aea0fb859ea0fb/../../../../../blobstore/13b711e5117ca3ad3c459a303dedfdb6dc0ad7c1440aabf02df6407a43bd94df/kernel-devel-4.14.299-223.520.amzn2.x86_64.rpm"
```

Reply:
So, they are stored in different blob stores that's why just updating the name is not working.
I'll share the small patch with you, basically `to_driverkit_config` should become
```python
def to_driverkit_config(self, release, deps):
    return [repo.DriverKitConfig(release, "amazonlinux{2,2022,2023}", dep) for dep in deps]
```

for each amazonlinux supported version.
I'll attach the patch [here](https://github.com/falcosecurity/libs/files/11653099/patch.txt).

Reply:
Works like a charm, ty ❤️ !